src/lib/libast/comp/system.c

```diff
@@ -43,13 +43,14 @@
 extern int
 system(const char* cmd)
 {
-	char*	sh[4];
+	char*	sh[5];
 
 	if (!cmd)
 		return !eaccess(pathshell(), X_OK);
 	sh[0] = "sh";
 	sh[1] = "-c";
-	sh[2] = (char*)cmd;
-	sh[3] = 0;
+	sh[2] = "--";
```

Comment:
This breaks on FreeBSD:

```
$ sh -c -- 'echo hi'
echo hi: --: not found
```

See: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=220587

They have been refusing to fix it for years because their ports scripts are broken. :-/